test_suite/tests/derive.rs

```diff
@@ -76,7 +76,6 @@ fn struct_derive() {
     assert_type!(SelfTyped, self_typed_type);
 }
 
-// TODO: dp – no need to keep all of these tests; pick one and toss the rest.
```

Comment:
Why is there no need for the other tests? As it looks on the surface level they are all testing the feature under different scenario.

Reply:
I copied the tests from the other PR where there was a greater need for validating that no `PhantomData` items were allowed in different scenarios. 
In this PR all `PhantomData` members are allowed through and I think it's a bit of an overkill to assert that yes, Phantoms are allowed everywhere (unless skipped, which isn't available in this PR).